### DISCUSSION TITLE
Release of ROCm 5.2.0

### DISCUSSION THREAD

**Discussion by tpkessler** [rocm-arch/rocm-arch#830] [Announcements]
All packages in this repository have been updated to ROCm 5.2.0! Main features of this release include:

* All packages build in a clean chroot
* Extended support for Big Navi
* New links to documentation as this moved to https://docs.amd.com/

Enjoy this new release, either by compiling from source or use the binary packages provided by `arch4edu`!

**Comment by acxz** [comment]
# Progress & Roadmap

I'd like to also provide an update on the "state of the union":
Previous: https://github.com/rocm-arch/rocm-arch/discussions/758

### Updates

1. push for local archlinux patches to be merged into upstream rocm packages
    We've been doing a really good job on this, all I can say is we keep this up! I'd provide a summary, but its honestly a decent size list. (I may do it at one point so that we have a list that we can point to and be proud about)

2. provide binary packages
    @petronny at https://github.com/arch4edu/arch4edu is back, baby! With a new web ui to boot as well: https://arch4edu.github.io/

3. inclusion of packages in the official Arch Linux repos
    See discussion https://github.com/rocm-arch/rocm-arch/discussions/673
    Looks like our very own @tpkessler will soon be a TU (thx @ArchangeGabriel & @lordheavy!) and be the liaison between these packages and the Archlinux team! We'll still keep this repo, but the sync process for an ever growing list of packages will change from the AUR to [community]. This will also mean that @petronny will have less work haha.

4. try to get these packages to be [reproducible](https://reproducible-builds.org/)
    To be honest, this is more the responsibility of upstream ROCm, and as a distributor of ROCm we shouldn't worry about this. Taking this off our vision/roadmap.

5. get higher level libraries working on top of this
    Some good work has been done here as well. python-pytorch-rocm, hipmagma, and hipsycl all work and we are slowly fixing the issues with tensorflow-rocm as well. Want to give a shout out to @wuxxin for adding the package for python-torchvision-rocm and testing other packages in the pytorch ecosystem like python-torchtext with ROCm. Also would like to thank @jeffdaily for reviewing my PRs over at pytorch and making this possible. Getting rocm support in the [community] blender is also on the list thanks to @svenstaro!

6. Get other rocm projects packages
    I'd say most relevant packages are packaged. Many of the remaining ones are simply meta-packages, which just increases maintainership burden when we need to update the entire stack. This is not a priority for us, unless there is a specific request from the community. See: https://github.com/rocm-arch/rocm-arch/issues/86 for more info

7. Get cuda backend hip packages packaged
    This is sorta related to 6). This one's on me... Not really any progress, yet. Shouldn't be hard, just have been lazy about it. See: https://github.com/orgs/rocm-arch/discussions/753 for more info about this

### New things!

8. Build unsupported gfx architectures
    A huge gripe (maybe the largest one I've seen) is the lack of ROCm support on newer (and older) AMD GPUs. Previously, I have been adamant on keeping the support on rocm-arch to be the same as official ROCm, mostly because I didn't want the rocm-arch community to be seen as annoying to upstream ROCm by submitting bug reports on configurations that ROCm doesn't support anyway. However, now I believe that as a packager and distributor of upstream ROCm code we should try to support as many configurations that can build. The bugs found can eventually help upstream in adding support for the new architectures. So please feel free to try building with new architectures and reporting/fixing issues! We'll be receptive as much as we can.

    Users can now specify what target to build their packages with the `AMDGPU_TARGETS` env variable. For binary builds from arch4edu, the upstream targets are used, but when building these packages from source you can specify which target to build with.

    In lieu of this gfx80* (Polaris) support is available, unlike upstream. I believe you can also build gfx1030 up to python-pytorch-rocm, gfx1031/gfx1032 still need work (however, upstream patches are available). I'd like to shout out @cgmb over at AMD for being very supportive of this endeavor.

### Other ROCm communities
The [Debian team](https://salsa.debian.org/rocm-team/community/team-project) has been working on the build systems and other improvements that have made our lives here easier. Shout out to @Maxzor and the other Debian-AI members.

There have also been some commits upstream by the [Gentoo folks](https://github.com/justxi/rocm) that have benefited us as well so I'd like to give some applause out to the efforts led by @justxi.

**Comment by tpkessler** [comment]
# Patch Release

All packages have been updated to 5.2.1. There's also a new package, `rocwmma`.
